Tipster/MainWindow.cpp

```diff
@@ -102,35 +102,39 @@ MainWindow::MessageReceived(BMessage* msg)
 			about->Show();
 			break;
 		}
-		case UPDATE_ICON:
+		case NEXT_TIP:
 		{
-			status_t status = msg->FindString("url", fURL);
-
-			if (status == B_OK) {
-				BString artwork;
-				msg->FindString("artwork", &artwork);
-
-				size_t size;
-				const uint8* iconData = (const uint8*)
-					fResources->LoadResource('VICN', artwork.String(),
-					&size);
-
-				if (size > 0) {
-					fIconBitmap = new BBitmap(BRect(0, 0, 64, 64), 0,
-						B_RGBA32);
-
-					status_t iconStatus = BIconUtils::GetVectorIcon(
-						iconData, size, fIconBitmap);
-
-					if (iconStatus == B_OK)
-						fIcon->SetIcon(fIconBitmap);
-				}
-			}
+			fTipsterView->UpdateTip();
+			break;
+		}
+		case PREVIOUS_TIP:
+		{
+			fTipsterView->DisplayPreviousTip();
+			break;
+		}
+		case DELAY_30_S:
```

Comment:
Doesn't this code feel a bit repetitive?
I suggest you have a look at BMessage->AddInt32 and BMessage->FindInt32, which will allow you to use a single message constant with different values.